src/EnergyPlus/HybridModel.cc

```diff
@@ -142,29 +166,221 @@ namespace HybridModel {
+                        }
+                    }
+
+                    // Scenario 2-3: To solve infiltration rate and people count
+                    if (!FlagHybridModel_TM && FlagHybridModel_AI && FlagHybridModel_PC) {
+
+                        // !!! Consider: which variable to use?
 
-                    // Zone Air Infiltration Rate and Zone Internal Thermal Mass calculations cannot be performed simultaneously
-                    if (HybridModelZone(ZonePtr).InternalThermalMassCalc && HybridModelZone(ZonePtr).InfiltrationCalc) {
-                        HybridModelZone(ZonePtr).InfiltrationCalc = false;
+                        // Temperature and humidity ratio are measured
+                        if (TemperatureSchPtr > 0 && HumidityRatioSchPtr > 0){
+                            HybridModelZone(ZonePtr).InfiltrationCalc_T = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_T = true;
+                            HybridModelZone(ZonePtr).InfiltrationCalc_H = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_H = true;
+                        }
+                        // Temperature and CO2 concentration are measured
+                        if (TemperatureSchPtr > 0 && CO2ConcentrationSchPtr > 0) {
+                            HybridModelZone(ZonePtr).InfiltrationCalc_T = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_T = true;
+                            HybridModelZone(ZonePtr).InfiltrationCalc_C = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_C = true;
+                        }
+                        //  Humidity ratio and CO2 concentration are measured
+                        if (HumidityRatioSchPtr > 0 && CO2ConcentrationSchPtr > 0) {
+                            HybridModelZone(ZonePtr).InfiltrationCalc_H = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_H = true;
+                            HybridModelZone(ZonePtr).InfiltrationCalc_C = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_C = true;
+                        }
+                    }
+
+					// Scenario 3: Three unknown parameters to solve
+					if (FlagHybridModel_TM && FlagHybridModel_AI && FlagHybridModel_PC) {
+						if (TemperatureSchPtr > 0 && HumidityRatioSchPtr > 0 && CO2ConcentrationSchPtr > 0) {
+                            HybridModelZone(ZonePtr).InternalThermalMassCalc_T = false;
+                            HybridModelZone(ZonePtr).InfiltrationCalc_T = true;
+                            HybridModelZone(ZonePtr).InfiltrationCalc_H = true;
+                            HybridModelZone(ZonePtr).InfiltrationCalc_C = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_T = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_H = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_C = true;
+						}
+					}
+
```

Comment:
The logic from Line #195 to Line #326 is confusing, and there's no input checking. Can be modified as followed based on my understanding. Please correct me if anything missing.
```
If solve internal mass:
	if has measured temperature (pointer > 0):
		if schedule is defined (pointer found / valid):
			InternalThermalMassCalc_T = true;
		else:
			show shedule not found error
	else: 
		show can’t not solve error - temperature is required

If solve air infiltration:
	if has measured temperature (pointer > 0):
		if schedule is defined (pointer found / valid):
			InfiltrationCalc_T = true;
		else:
			show shedule not found error
	if has measured humidity (pointer > 0 and pointer valid ):
		...
	if has measured CO2 (pointer > 0 and pointer valid ):
		...
	else if none of them defined: 
		show can’t not solve error - either temperature, humidity or CO2 is required
```

Reply:
@tsbyq